model/component.js

```diff
@@ -298,20 +314,91 @@ class Component extends ParchKey {
      * @return {boolean}
      */
     validateFeature() {
-        let valid = super.validate();
-        valid = Validation.testStringValue(this.layer, 'layer', 'Component') ? valid : false;
+        let valid = true;
+
+        if (!this.component || !this.component.validate()) {
+            valid = false;
+            console.log('Component Feature: Field "component" is invalid.');
+        }
+
+        valid = this.validateFeatureLayer() ? valid : false;
+
         if (!this.location) {
             valid = false;
-            console.log('Component: Field "location" is invalid.');
+            console.log('Component Feature: Field "location" is invalid.');
         } else {
             valid = this.location.validate() ? valid : false;
         }
-        valid = this.validateSpans() ? valid : false;
-        valid = Validation.testDimensionValue(this.depth, 'depth', 'Component') ? valid : false;
+
+        valid = this.validateFeatureSpans() ? valid : false;
+        valid = Validation.testDimensionValue(this.depth, 'depth', 'Component Feature') ? valid : false;
+
+        return valid;
+    }
+
+    /**
+     * Validate the x/y-spans in the scope of a Component Feature.
+     *
+     * Both x- and y-spans must be valid. Both x- and y- spans must match the
+     * abstract Component's spans.
+     *
+     * @since 1.0.0
+     *
+     * @see Validation.testSpanValue
+     *
+     * @returns {boolean}
+     */
+    validateFeatureSpans() {
+        let valid = this.validateSpans();
+
+        // Only compare spans if the component is referenced
+        if (this.component) {
+            if (this.xSpan !== this.component.xSpan) {
+                valid = false;
+                console.log('Component Feature: Field "xSpan" (' + this.xSpan + 'does not match the abstract' +
+                        ' Component\'s xSpan (' + this.component.xSpan + '.');
+            }
+            if (this.ySpan !== this.component.ySpan) {
+                valid = false;
+                console.log('Component Feature: Field "ySpan" (' + this.ySpan + 'does not match the abstract' +
+                        ' Component\'s ySpan (' + this.component.ySpan + '.');
+            }
+        }
```

Comment:
Should we also check that the `xSpan` and `ySpan` values are > 0 here?

Reply:
I assumed we'd catch that in the abstract Component, but I added it in just to be sure.